pygam/pygam.py

```diff
@@ -87,22 +87,74 @@
              'diffs': Diffs,
              'accuracy': Accuracy,
              'coef': Coef
-            }
+             }
 
 PENALTIES = {'auto': 'auto',
              'derivative': derivative,
              'l2': l2,
              'none': none,
-            }
+             }
 
 CONSTRAINTS = {'convex': convex,
                'concave': concave,
                'monotonic_inc': monotonic_inc,
                'monotonic_dec': monotonic_dec,
                'circular': circular,
                'none': none
-              }
-
+               }
+
+
+def _parallel_inc(gam, X, y, weights):
```

Comment:
why did you prefer to make this a function instead of a method of the GAM?

Reply:
@maorn

Reply:
all objects that you pass in multiprocess (apply_async) should be  pickable (serialized, see here,
https://docs.python.org/3/library/pickle.html#what-can-be-pickled-and-unpickled) 
when you add multiprocess into a class there can be some issues with class members, so in general i always write the multiprocess part as an external function. 
you can insert it back and check that it works

Reply:
cool. i suspect that it should be OK as a method. 

since the argument `gam` would be the same as the argument `self` ?

Reply:
yes it should be fine, mind that all the classes that inherent from GAM class should be pickable as well to run this function